dev/src/sdlaudiosfxr.cpp

```diff
@@ -422,14 +424,30 @@ void SDLSoundClose() {
     SDL_QuitSubSystem(SDL_INIT_AUDIO);
 }
 
-int SDLPlaySound(string_view filename, bool sfxr, float vol, int loops) {
+int SDLPlaySound(string_view filename, bool sfxr, float vol, int loops, int prio) {
     if (!SDLSoundInit()) return 0;
-
     auto snd = LoadSound(filename, sfxr);
     if (!snd) return 0;
-
-    int ch = Mix_PlayChannel(-1, snd->chunk.get(), loops);
-    if (ch >= 0) Mix_Volume(ch, (int)(MIX_MAX_VOLUME * vol)); // set channel to default volume (max)
+    if (prio < 0) prio = 0;
+    else if (prio > 2) prio = 2;
+    int ch = Mix_GroupAvailable(-1); // is there any free channel?
+    if (ch == -1) { // no free channel?
+        // delete the oldest sound with the lowest possible priority
+        int ch_try = -1;
+        for (int p = 0; p <= prio; p++) {
+            ch_try = Mix_GroupOldest(p);
+            if (ch_try > -1) {
+               Mix_HaltChannel(ch_try); 
+               ch = ch_try;
+               break;
+            }
+        }
+    }
+    if (ch > -1) { // ch should never be -1 at this point anyhow
```

Comment:
`ch >= 0`

ch might be -1 here if you tried to play something with normal prio, and all channels are taken up by higher prio groups.